internal/konnect/tracing/datadog.go

```diff
@@ -0,0 +1,61 @@
+package tracing
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/logging"
+)
+
+// DoRequest is a helper function that sends an HTTP request and logs the result with DataDog trace ID.
+func DoRequest(ctx context.Context, httpClient *http.Client, req *http.Request) (*http.Response, error) {
```

Comment:
Might be worth spending a moment to write a handful of UTs for this `func`, WDYT?